Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt

```diff
@@ -2273,7 +2275,7 @@ namespace FluentAssertions.Specialized
         protected FluentAssertions.AndConstraint<TAssertions> NotThrowInternal(System.Exception exception, string because, object[] becauseArgs) { }
         protected FluentAssertions.AndConstraint<TAssertions> NotThrowInternal<TException>(System.Exception exception, string because, object[] becauseArgs)
             where TException : System.Exception { }
-        protected FluentAssertions.Specialized.ExceptionAssertions<TException> ThrowInternal<TException>(System.Exception exception, string because, object[] becauseArgs)
+        protected FluentAssertions.Specialized.ExceptionAssertions<TException> ThrowInternal<TException>(System.Exception exception, System.TimeSpan timeSpan, string because, object[] becauseArgs)
```

Comment:
We can avoid changing this overload, which I would also prefer since `InvokeWithInterceptionAsync(Func<Task> action, TimeSpan timeSpan)` only uses `action` in the true part and `timeSpan` in the false part.

https://github.com/jnyrup/fluentassertions/commit/9f5676cd7015674d62313a2c1a2239b7e07b6ea5

Reply:
I don't understand what @jnyrup is suggesting, but I would just duplicate the implementation of `ThrowInternal`. In retrospect, it should have never been added to the base-class and should have been a local `private` method. IMO, base-classes should never be used to reuse code. Use composition instead.

Reply:
I'm suggesting the same. 
Duplicating `ThrowInternal` and `InvokeWithInterceptionAsync` rather than introducing conditional logic to fit both flows into the same methods.

Reply:
I'm fine with optimizing the code, but I would like to understand your motivations.

Dennis wrote "IMO, base-classes should **never** be used to reuse code."
Why you are such strict? What is your intention for base-classes?

Jonas wrote "... and timeSpan (only) in the false part" which is not correct because it is part of the condition,
and "... rather than introducing conditional logic to fit both flows into the same methods." without any explanation.

I think having one method `InvokeWithInterceptionAsync` including some variations is easier to understand than two methods with same name and totally different signature and behavior (more than common for method overloads).

Reply:
Let's just say that I've experienced the pain more than a few times. Using base-classes like that leads to classes that don't have a clear purpose other than reusing logic. This often creates unnecessary coupling that I'd like to avoid. I've seen my fair share of those becoming god classes that nobody really understands. In this case, if we would have kept the logic private to the consuming class, we would have never had a need for the 'protected` member in the first place and never had to deal with the potentially breaking change.

Reply:
Then please define and check such rules for FA globally. 
We have already several base classes with and without protected members.

Reply:
We were discussing this in the context of this specific PR. And those rules are not that black and white. Sometimes a base class is fine and sometimes it isn't. It all depends on its purpose. Also, time will also change the way we look at our own code.

Reply:
But this makes it complicated for contributors. 
Things I did in the "right way" in the past are "wrong" next time.

Reply:
Yep, that's possible. And that's why we review all PRs ;-)

Reply:
I've restarted the PR with all the received input.
Please review again.